rotkehlchen/assets/utils.py

```diff
@@ -0,0 +1,46 @@
+import logging
+from typing import Optional, Union
+
+from rotkehlchen.errors import UnknownAsset
+from rotkehlchen.typing import ChecksumEthAddress
+
+from .asset import EthereumToken
+from .unknown_asset import UnknownEthereumToken
+
+
+log = logging.getLogger(__name__)
+
+
+def get_ethereum_token(
+        symbol: str,
+        ethereum_address: ChecksumEthAddress,
+        name: Optional[str] = None,
+        decimals: Optional[int] = None,
+) -> Union[EthereumToken, UnknownEthereumToken]:
+    """Given a token symbol and address return the <EthereumToken>, otherwise
+    an <UnknownEthereumToken>.
+    """
+    ethereum_token = None
+    is_unknown_asset = False
+
+    try:
+        ethereum_token = EthereumToken(symbol)
+    except UnknownAsset:
+        is_unknown_asset = True
+    else:
+        if ethereum_token.ethereum_address != ethereum_address:
+            is_unknown_asset = True
+
+    if is_unknown_asset:
+        log.error(
+            f'Encountered unknown asset {symbol} with address '
+            f'{ethereum_address}. Instantiating UnknownEthereumToken',
+        )
+        ethereum_token = UnknownEthereumToken(  # type: ignore
```

Comment:
Why the type ignore here? It's not needed.

Replace 

`ethereum_token = None` in line 23 with `ethereum_token: Union[EthereumToken, UnknownEthereumToken]` to declare the type.

And then it should work. I may be wrong :smile: it's late. But try it out.

Reply:
I'm new to typing... didn't know that by defining the type I could avoid referencing a value!

I tried many times `ethereum_token: Union[EthereumToken, UnknownEthereumToken] = None` and `ethereum_token: Union[EthereumToken, UnknownEthereumToken, None] = None`, lol